src/common/getters.ts

```diff
@@ -775,6 +436,321 @@ export function getInherit(vc: DatasetWithId): boolean {
+ *
+ * @hidden
+ */
+function deserializeFields<T>(
+  vc: DatasetWithId,
+  field: URL,
+  deserializer: (value: Literal) => T | undefined,
+  type: string,
+): T[] {
+  return Array.from(
+    vc.match(getConsent(vc), namedNode(field.href), null, defaultGraph()),
+  )
+    .map((q) => {
+      if (q.object.termType !== "Literal") {
+        throw new Error(
+          `Expected value object for predicate ${field.href} to be a litteral, found ${q.object.termType}.`,
+        );
+      }
+      return q.object as Literal;
+    })
+    .map((object) => {
+      const result = deserializer(object);
+      if (result === undefined) {
+        // FIXME use inrupt error library
+        throw new Error(
+          `Failed to deserialize value ${object.value} for predicate ${field.href} as type ${type}.`,
+        );
+      }
+      return result;
+    });
+}
+
+function deserializeField<T>(
+  vc: DatasetWithId,
+  field: URL,
+  deserializer: (value: Literal) => T | undefined,
+  type: string,
+): T | undefined {
+  const result = deserializeFields(vc, field, deserializer, type);
+  if (result.length > 1) {
+    // FIXME use inrupt error library
+    throw new Error(
+      `Expected one value for predicate ${field.href}, found many: ${result}`,
+    );
+  }
+  return result[0];
+}
+
+const xmlSchemaTypes = {
+  boolean: namedNode("http://www.w3.org/2001/XMLSchema#boolean"),
+  double: namedNode("http://www.w3.org/2001/XMLSchema#double"),
+  integer: namedNode("http://www.w3.org/2001/XMLSchema#integer"),
+  string: namedNode("http://www.w3.org/2001/XMLSchema#string"),
+} as const;
+
+function deserializeBoolean(serialized: Literal): boolean | undefined {
+  if (!serialized.datatype.equals(xmlSchemaTypes.boolean)) {
+    return undefined;
+  }
+  if (serialized.value === "true") {
+    return true;
+  }
+  if (serialized.value === "false") {
+    return false;
+  }
+  return undefined;
+}
+
+/**
+ * Reads the custom boolean value with the provided name in the consent section of the provided Access Credential.
+ * Throws on type mismatch, and returns `undefined` if no values are found.
+ *
+ * @example
+ * ```
+ * const accessRequest = await issueAccessRequest({...}, {
+ *  ...,
+ *  customFields: new Set([
+ *     {
+ *       key: new URL("https://example.org/ns/customBoolean"),
+ *       value: true,
+ *     },
+ *   ]),
+ * });
+ * // b is true
+ * const b = getCustomBoolean(accessRequest, new URL("https://example.org/ns/customBoolean"));
+ * ```
+ *
+ * @param accessCredential The Access Credential (Access Grant or Access Request)
+ * @returns the custom boolean field with the provided name
+ * @since unreleased
+ */
+export function getCustomBoolean(
+  accessCredential: DatasetWithId,
+  field: URL,
+): boolean | undefined {
+  return deserializeField(
+    accessCredential,
+    field,
+    deserializeBoolean,
+    "boolean",
+  );
+}
+
+function deserizalizeDouble(serialized: Literal): number | undefined {
```

Comment:
Any reason this wasn't renamed to `deserializeFloat`?

Reply:
That's a copy-paste error, or a search-and replace race condition if you will

Reply:
Fixed with ff9c00d